ui/app/routes/workspace/projects/project/app/deployment/deployment-seq.ts

```diff
@@ -32,16 +37,46 @@ export default class DeploymentDetail extends Route {
   }
 
   async model(params: Params): Promise<Model> {
-    let { deployments } = this.modelFor('workspace.projects.project.app') as AppRouteModel;
+    let { builds, deployments, releases } = this.modelFor('workspace.projects.project.app') as AppRouteModel;
     let deployment = deployments.find((obj) => obj.sequence == Number(params.sequence));
+    let build = builds.find((obj) => obj.pushedArtifact.id === deployment?.pushedArtifact.id);
 
     if (!deployment) {
       throw new Error(`Deployment v${params.sequence} not found`);
     }
 
     let deploymentId = deployment.id;
-    let { releases } = this.modelFor('workspace.projects.project.app') as AppRouteModel;
     let release = releases.find((r) => r.deploymentId === deploymentId);
-    return { ...deployment, release };
+
+    let timeline: TimelineModel = {};
+    if (build) {
+      timeline.build = {
+        sequence: build.sequence,
+        status: build.status,
+      };
+    }
+
+    timeline.deployment = {
+      sequence: deployment.sequence,
+      status: deployment.status,
+    };
+
+    if (release) {
+      timeline.release = {
+        sequence: release.sequence,
+        status: release.status,
+      };
+    }
+
+    return { ...deployment, release, timeline };
+  }
+
+  resetController(_: Controller, isExiting: boolean, transition: Transition): void {
```

Comment:
Fixes a bug if you clicked on the 'deployments' tab when already being there, you get redirected to a blank page.

Reply:
Good catch!